frontend/js/src/user/UserTaste.tsx

```diff
@@ -86,6 +86,7 @@ export default class UserTaste extends React.Component<UserTasteProps> {
           listenBrainzAPIBaseURI={APIService.APIBaseURI}
           refreshSpotifyToken={APIService.refreshSpotifyToken}
           refreshYoutubeToken={APIService.refreshYoutubeToken}
+          refreshSoundcloudToken={APIService.refreshSoundcloudToken}
```

Comment:
This is getting really annoying, isn't it?
Since each of those functions is just calling the APIService method with the name of the service as a string, I think we should be able to refactor it (separate PR, I don't mind getting in there and doing it) and just pass the `refreshAccessToken` method to BrainzPlayer.

Then in BrainzPlayer we can pass the prop along, for example 
```
<SoundcloudPlayer
  ...
  refreshToken={refreshAccessToken}
```

And finally, in SoundcloudPlayer, we can call `this.props.refreshToken(this.name)` <- the name string is already stored in the class.


Anyway, I'll have a look at refactoring that once this PR is merge, so as not to be disruptive :)

Reply:
Makes sense